shell/platform/darwin/ios/framework/Source/FlutterDartSource.h

```diff
@@ -13,14 +13,15 @@ typedef void (^ValidationResult)(BOOL result, NSString* message);
 
 @property(nonatomic, readonly) NSURL* dartMain;
 @property(nonatomic, readonly) NSURL* packages;
-@property(nonatomic, readonly) NSURL* flxArchive;
+@property(nonatomic, readonly) NSURL* flutterAssets;
```

Comment:
This is modifying a header that is published as part of the `Flutter.framework`. Existing users of the framework (iOS embedders) will break if they are currently using this API.  We should decide whether to keep this API as-is and have the initializer return `nil` when queried with FLX assets for the sake of backwards header compatibility. We don't want to get into a situation where a Flutter upgrade causes the developer's embedder to break. I would land this patch once all the changes that remove FLXs are consolidated and we can make one breaking change along with notifications on appropriate channels and changelogs.

Reply:
The change above is not in the public API, however the change in FlutterDartProject.h is, and I agree with postponing the breaking change until FLX is completely removed. I'll change this PR to keep the FLX initializer and add an initializer with Flutter assets directory, WDYT?

Reply:
This will be a breaking change once we stop putting assets in app.flx anyway, so we might as well break it (with the right notices, of course) once.

Since no-one reads notices ahead of time anyway, I think I'd prefer to find out about this by having my build break up front, rather than having the build work but assets mysteriously no longer appear in my app. :)

Reply:
> I'll change this PR to keep the FLX initializer and add an initializer with Flutter assets directory, WDYT?

Sounds good to me.

> This will be a breaking change once we stop putting assets in `app.flx` anyway, so we might as well break it (with the right notices, of course) once.

Since we will be updating the engine and the supporting tooling (that generates the FLX files and configures application launch) in one go, there should be no runtime misbehavior due to the embedder looking for, but not finding the FLX in a particular spot. However, embedders using the old API will break even though those APIs will never be invoked by the application launch mechanism.

> Since no-one reads notices ahead of time anyway, I think I'd prefer to find out about this by having my build break up front,

A kinder, more tactful, approach in dealing with this update would be to decorate the old API with `NS_DEPRECATED` attributes with a clear message regarding the direction we are going to take moving forward and what the user should do to rectify the warning. Adroit users already have warnings set as errors and will be able to handle the update on their own (especially if we indicate that the old API is a no-op that the application launch mechanism wont invoke anyway). Users who are perhaps less familar with the native side of mobile development and build systems wont have their builds break on them due to a routine update.

Reply:
You mentioned a doc with instructions on how this should be done. Are you able to find it? If not, could you please let me know more specifically how to do the tagging, and the update of the changelog etc. ?